sdk/core/Azure.Core/src/Pipeline/Policies/RequestActivityPolicy.cs

```diff
@@ -0,0 +1,117 @@
+
+        public override Task ProcessAsync(HttpPipelineMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline)
+        {
+            return ProcessAsync(message, pipeline, true);
+        }
+
+        public override void Process(HttpPipelineMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline)
+        {
+            ProcessAsync(message, pipeline, false).EnsureCompleted();
+        }
+
+        private static async Task ProcessAsync(HttpPipelineMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline, bool isAsync)
+        {
+            if (!s_diagnosticSource.IsEnabled())
+            {
+                await ProcessNextAsync(message, pipeline, isAsync).ConfigureAwait(false);
+
+                return;
+            }
+
+            var activity = new Activity("Azure.Core.Http.Request");
+            activity.AddTag("http.method", message.Request.Method.Method);
+            activity.AddTag("http.url", message.Request.UriBuilder.ToString());
+            if (message.Request.Headers.TryGetValue("User-Agent", out string userAgent))
+            {
+                activity.AddTag("http.user_agent", userAgent);
+            }
+
+            var diagnosticSourceActivityEnabled = s_diagnosticSource.IsEnabled(activity.OperationName, message);
+
+            if (diagnosticSourceActivityEnabled)
+            {
+                s_diagnosticSource.StartActivity(activity, message);
+            }
+            else
+            {
+                activity.Start();
+            }
+
+
+            if (isAsync)
+            {
+                await ProcessNextAsync(message, pipeline).ConfigureAwait(false);
+            }
+            else
+            {
+                ProcessNext(message, pipeline);
+            }
+
+            activity.AddTag("http.status_code", message.Response.Status.ToString(CultureInfo.InvariantCulture));
+
+            if (diagnosticSourceActivityEnabled)
+            {
+                s_diagnosticSource.StopActivity(activity, message);
+            }
+            else
+            {
+                activity.Stop();
+            }
+        }
+
+        private static async Task ProcessNextAsync(HttpPipelineMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline, bool isAsync)
+        {
+            var currentActivity = Activity.Current;
+
+            if (currentActivity != null)
+            {
+                if (currentActivity.IdFormat == ActivityIdFormat.W3C)
+                {
+                    if (!message.Request.Headers.Contains(TraceParentHeaderName))
+                    {
+                        message.Request.Headers.Add(TraceParentHeaderName, currentActivity.Id);
+                        if (currentActivity.TraceStateString != null)
+                        {
+                            message.Request.Headers.Add(TraceStateHeaderName, currentActivity.TraceStateString);
+                        }
+                    }
+                }
+                else
```

Comment:
I think supporting Request-Id doesn't make sense. Azure Services will never implement Request-Id header so propagating it won't help anything.

Reply:
So what should we do in case of non-W3C activity?

Reply:
Two options
1. Insist on W3C activity (in preview7 there will be an instance method Activity.SetIdFormat()). Then if a user had activity in the old format, the one created by Azure SDK will not correlate to the old one. But anything that happens in and after Azure SDK (e.g. EventHub and message processing or Storage logs and EventGrid notifications followed by Function)  will be correlated.

2. Tolerate old format but not inject headers. Then everything before and including Azure SDK will correlate, anything after will become a new trace.

Looking into this, there is no good solution for format mismatch. But I think this is not Azure SDK responsibility to solve this issue - tracing system should control the format and enforce W3C on the incoming boundary. 

In the bright future, there will be nothing else than W3C and we won't have this problem. 

So I suggest do 2.